hw/drivers/sensors/bma253/src/bma253.c

```diff
@@ -2942,14 +3266,16 @@ reset_and_recfg(struct bma253 * bma253)
         return rc;
     }
 
-    fifo_cfg.fifo_mode = FIFO_MODE_BYPASS;
+    fifo_cfg.fifo_mode = FIFO_MODE_STREAM;
```

Comment:
Do not change this to `STREAM`, the basic should always be polling mode and hence the mode should be set to `BYPASS`

Reply:
Hello
this is a small optimization we tried to make over it.
in polling mode, it does not hurt to set the fifo mode to STREAM, because the data is polled from the data registers and it's not affected by the fifo settings.
in stream mode, we need the fifo_mode to be set to FIFO_MODE_STREAM, and if it's already setup in this way (in reset_and_recfg), we don't need to setup it again at time of stream_read, this saves a bit of time for the driver.

But I agree that this is not huge optimization, so if you think it's necessary to keep the original setting, we can do that.